packages/blocks/src/paragraph-block/paragraph-block.ts

```diff
@@ -123,6 +126,13 @@ export class ParagraphBlockComponent extends BlockComponent<
       }
       return true;
     });
+
+    this._isReadonly = this.model.doc.readonly;
+    this.model.doc.awarenessStore.slots.update.on(() => {
+      if (this._isReadonly !== this.model.doc.readonly) {
+        this._isReadonly = this.model.doc.readonly;
+      }
+    });
```

Comment:
@Saul-Mirone Maybe implementing this with a signal-based method would be better?